ceph.spec.in

```diff
@@ -792,28 +757,12 @@ Group:		Development/Libraries/Python
 Requires:	librgw2 = %{_epoch_prefix}%{version}-%{release}
 Requires:	python%{python3_pkgversion}-rados = %{_epoch_prefix}%{version}-%{release}
 %{?python_provide:%python_provide python%{python3_pkgversion}-rgw}
-%if 0%{without python2}
 Provides:	python-rgw = %{_epoch_prefix}%{version}-%{release}
```

Comment:
i don't think it makes sense to `Provide` python-* packages in a pure python3 env.

Reply:
I'm a bit confused here because you also mention that `Provides:  python-rbd ...` should stay.  Which one is it? Or is it a subset of the packages?

Reply:
i searched in this PR, and failed to find 

> Provides: python-rbd ... should stay. 

i think my argument here is 

1. we should let python3-* provide python-*.
1. since we don't have python2 packages, we can just drop python-*.

Reply:
I am having trouble following this. If we let python3- provide python- how can I drop python-* at the same time? 

In this specific line:

```
%if 0%{without python2}
Provides:	python-rgw = %{_epoch_prefix}%{version}-%{release}
```

Should I: 

1.  Remove it completely
2. Leave it as-is but remove the `without python2` section

Can I apply the same to *all* the `Provides: python-*` lines?

Reply:
Remove the surrounding `%if 0%{without py thon2}` and `%endif`, but leave the inside as-is.

In other words, the expression `0%{without python2}` can be assumed to be true always. The semantic meaning of it was "This is an OS where Python 3 is a first-class citizen, and the default Python version."

Reply:
And yes, you can apply this to all the `Provides: python-*` lines that are enclosed in such an if clause.